python312Packages.xlwt: Remove nose dependency

pyrox0 opened the pull request:
## Description of changes
Part of #326513
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

pyrox0 committed:
python312Packages.xlwt: Remove nose dependency

emilazy reviewed (commented):
Thanks. I think you can remove the `pythonOlder` as we don’t have any versions remotely that old.

Since the repository is archived and there are [some miscellaneous fix commits](https://github.com/python-excel/xlwt/compare/1.3.0...master)  (including removing nose) on GitHub postdating the final tag, maybe we should consider pinning the final commit?

pyrox0 committed:
python312Packages.xlwt: 1.3.0 -> 1.3.0-unstable-2018-09-16

Updates to the latest commit of the now-archived repo, should have no
breaking changes, per the suggestion in
https://github.com/NixOS/nixpkgs/pull/330217#pullrequestreview-2202431845

emilazy commented on pkgs/development/python-modules/xlwt/default.nix:
```diff
@@ -1,49 +1,41 @@
+buildPythonPackage {
   pname = "xlwt";
-  version = "1.3.0";
+  version = "unstable-2018-09-16";
```
```suggestion
  version = "1.3.0-unstable-2018-09-16";
```

(Not sure if this is documented in the manual yet but we agreed to include the previous release version in `unstable` versions, or `0-unstable-…` if there is none.)

pyrox0 replied:
Wasn't aware of that, will update the version.

emilazy commented on pkgs/development/python-modules/xlwt/default.nix:
```diff
@@ -1,49 +1,41 @@
     description = "Library to create spreadsheet files compatible with MS";
     homepage = "https://github.com/python-excel/xlwt";
-    license = with licenses; [
+    license = with lib.licenses; [
       bsdOriginal
       bsd3
       lgpl21Plus
     ];
```
Crazy ass moments in FOSS licensing.

emilazy reviewed (approved):
Result of `nixpkgs-review pr 330217` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>57 packages built:</summary>
  <ul>
    <li>netbox (netbox_3_7)</li>
    <li>odoo</li>
    <li>odoo.dist</li>
    <li>odoo16</li>
    <li>odoo16.dist</li>
    <li>peering-manager</li>
    <li>python311Packages.canmatrix</li>
    <li>python311Packages.canmatrix.dist</li>
    <li>python311Packages.canopen</li>
    <li>python311Packages.canopen.dist</li>
    <li>python311Packages.django-import-export</li>
    <li>python311Packages.django-import-export.dist</li>
    <li>python311Packages.django-tables2</li>
    <li>python311Packages.django-tables2.dist</li>
    <li>python311Packages.netbox-reorder-rack</li>
    <li>python311Packages.netbox-reorder-rack.dist</li>
    <li>python311Packages.pyexcel-ods</li>
    <li>python311Packages.pyexcel-ods.dist</li>
    <li>python311Packages.pyexcel-xls</li>
    <li>python311Packages.pyexcel-xls.dist</li>
    <li>python311Packages.pylint-django</li>
    <li>python311Packages.pylint-django.dist</li>
    <li>python311Packages.pytablewriter</li>
    <li>python311Packages.pytablewriter.dist</li>
    <li>python311Packages.pytest-md-report</li>
    <li>python311Packages.pytest-md-report.dist</li>
    <li>python311Packages.riscof</li>
    <li>python311Packages.riscof.dist</li>
    <li>python311Packages.riscv-isac</li>
    <li>python311Packages.riscv-isac.dist</li>
    <li>python311Packages.xlwt</li>
    <li>python311Packages.xlwt.dist</li>
    <li>python312Packages.canmatrix</li>
    <li>python312Packages.canmatrix.dist</li>
    <li>python312Packages.canopen</li>
    <li>python312Packages.canopen.dist</li>
    <li>python312Packages.django-import-export</li>
    <li>python312Packages.django-import-export.dist</li>
    <li>python312Packages.django-tables2</li>
    <li>python312Packages.django-tables2.dist</li>
    <li>python312Packages.netbox-reorder-rack</li>
    <li>python312Packages.netbox-reorder-rack.dist</li>
    <li>python312Packages.pylint-django</li>
    <li>python312Packages.pylint-django.dist</li>
    <li>python312Packages.pytablewriter</li>
    <li>python312Packages.pytablewriter.dist</li>
    <li>python312Packages.pytest-md-report</li>
    <li>python312Packages.pytest-md-report.dist</li>
    <li>python312Packages.riscof</li>
    <li>python312Packages.riscof.dist</li>
    <li>python312Packages.riscv-isac</li>
    <li>python312Packages.riscv-isac.dist</li>
    <li>python312Packages.xlwt</li>
    <li>python312Packages.xlwt.dist</li>
    <li>tandoor-recipes</li>
    <li>visidata</li>
    <li>visidata.dist</li>
  </ul>
</details>
